pyscript.core/docs/README.md

```diff
@@ -0,0 +1,481 @@
+
+  </div>
+</details>
+
+<details>
+  <summary><strong>Target</strong></summary>
+  <div>
+
+When it comes to *strings* or *attributes*, we consider the **target** any valid element's *id* on the page or, in most cases, any valid *CSS* selector.
+
+```html
+<!-- ℹ️ - requires py-script custom type -->
+<script type="py">
+    # target here is a string
+    display('Hello PyScript', target='output')
+</script>
+<div id="output">
+    <!-- will show "Hello PyScript" once the script executes -->
+</div>
+```
+
+When it comes to the `property` or `field` attached to a `<script>` element though, that *id* or *selector* would already be resolved, so that such field would always point at the very same related element.
+
+```html
+<script type="micropython" target="output">
+    from js import document
+    document.currentScript.target.textContent = "Hello";
+</script>
+<div id="output">
+    <!-- will show "Hello" once the script executes -->
+</div>
+```
+
+ℹ️ - Please note that if no `target` attribute is specified, the *script* will automatically create a "_companion element_" when the `target` property/field is accessed for the very first time:
+
+```html
+<script type="micropython">
+    from js import document
+
+    # will create a <script-micropython> element appended
+    # right after the currently executing script
+    document.currentScript.target.textContent = "Hello";
+</script>
+<!--
+    created during previous code execution
+
+    <script-micropython>Hello</script-micropython>
+-->
+```
+
+  </div>
+</details>
+
+<details>
+  <summary><strong>Env</strong></summary>
+  <div>
+
+ℹ️ - This is an **advanced feature** that is worth describing but usually it is not needed for most common use cases.
+
+Mostly due its terseness that plays nicely as attribute's suffix, among its commonly understood meaning, we consider an *env* an identifier that guarantee the used *interpreter* would always be the same and no other interpreters, even if they point at very same project, could interfere with globals, behavior, or what's not.
+
+In few words, every single *env* would spawn a new interpreter dedicated to such env, and global variables defined elsewhere will not affect this "_environment_" and vice-versa, an *env* cannot dictate what will happen to other interpreters.
+
+```html
+<!-- default env per each interpreter -->
+<script type="micropython">
+    shared = True
+</script>
+<script type="micropython">
+    # prints True - shared is global
+    print(shared)
+</script>
+
+<!-- dedicated interpreter -->
+<script type="micropython" env="my-project-env">
+    # throws an error - shared doesn't exist
+    print(shared)
+</script>
+```
+
+ℹ️ - Please note if the interpreter takes 1 second to bootstrap, multiple *environments* will take *that* second multiplied by the number of different environments, which is why this feature is considered for **advanced** use cases only and it should be discouraged as generic practice.
```

Comment:
It'd be good to add how those seconds add up here (how much can run in parallel and how much is serial)

Reply:
every *env* is a different pyodide, the only removed time with multiple envs is the time to download the module ... let's say 100ms ... if `loadPyodide` takes 1 second to bootstrap, there's no parallel in the same thread, every second will sum up to the amount of envs, the main thread is one, has no parallel work there with WASM modules running in the very same thread.